iothub_client/samples/pnp/pnp_temperature_controller/pnp_deviceinfo_component.c

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// The DTDL for this interface is defined at https://repo.azureiotrepository.com/Models/dtmi:azure:DeviceManagement:DeviceInformation;1?api-version=2020-05-01-preview
+
+
+// Standard C header files
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+// PnP helper routines
+#include "pnp_deviceinfo_component.h"
+#include "pnp_protocol_helpers.h"
+
+// Core IoT SDK utilities
+#include "azure_c_shared_utility/xlogging.h"
+
+// Property names along with their simulated values.  
+// NOTE: the property values must be legal JSON values.  Strings specifically must be enclosed with an extra set of quotes to be legal json string values.
+// The property names in this sample do not hard-code the extra quotes because the underlying PnP helper adds this to names automatically.
+#define PNP_ENCODE_STRING_FOR_JSON(str) "\"" str "\""
```

Comment:
how about using this 

#define stringfy(S) #S



Something like this
[https://onlinegdb.com/BkUc4pnkP](https://onlinegdb.com/BkUc4pnkP)

Reply:
Definitely want to use the preprocesser stringify `#`.  I'm going to leave the macro named a more verbose `PNP_ENCODE_STRING_FOR_JSON` so that the intent here is clear.  Fact you have to explicitly add the extra "..." business for JSON strings via a C string build up has caused some confusion so I'd want to be really explicit on that point.